File: 04.MedianOfTwoSortedArrays/main.js
var findMedianSortedArrays = function(nums1, nums2) {
    let mergedArray = nums1.concat(nums2).sort((a, b) => a - b);
    let total = mergedArray.length;
    if (total % 2 === 0) {
        return (mergedArray[total / 2 - 1] + mergedArray[total / 2]) / 2;
    } else {
        return mergedArray[Math.floor(total / 2)];
    }
};

console.log("Test 1: Median =", findMedianSortedArrays([1, 3], [2]));
console.log("Test 2: Median =", findMedianSortedArrays([1, 2], [3, 4]));
console.log("Test 3: Median =", findMedianSortedArrays([], [1]));
console.log("Test 4: Median =", findMedianSortedArrays([1, 2, 2], [2, 3, 4]));
console.log("Test 5: Median =", findMedianSortedArrays([-5, -3, -1], [2, 4, 6]));

File: 04.MedianOfTwoSortedArrays/main.py
from typing import List

class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        merged_array = sorted(nums1 + nums2)
        total = len(merged_array)
        if total % 2 == 0:
            mid1 = merged_array[total // 2 - 1]
            mid2 = merged_array[total // 2]
            return (mid1 + mid2) / 2.0
        else:
            return float(merged_array[total // 2])
